src/System.Linq.Expressions/src/System/Dynamic/Utils/CollectionExtensions.cs

```diff
@@ -15,22 +15,5 @@ public static T[] AddLast<T>(this IList<T> list, T item)
             res[list.Count] = item;
             return res;
         }
-
-        public static T First<T>(this IEnumerable<T> source)
-        {
-            var list = source as IList<T>;
-            if (list != null)
-            {
-                return list[0];
```

Comment:
This is where this helper differs from `Enumerable.First` which checks for a non-zero element count and prefers to consistently throw `InvalidOperationException` if the collection is empty.